github-actions/README.md

```diff
@@ -0,0 +1,87 @@
+# GitHub Actions
+
+## Benefits and options for Packit and users of Packit
+
+### Customization as a whole
+
+In case of providing Packit API via GitHub Actions it allows to run pipeline (SRPM → Copr →
+Testing Farm) after:
+
+- running some initial CI checks (saves resources on our side)
+- preparing archives (eliminates need for retries and failed jobs on our side, if they are blocked
+  by some external action)
+
+### SRPM builds
+
+- Allows user to **easily** customize the dependencies, i.e. does not require changes to be made to
+  the Sandcastle image.
+- Provides access to created SRPMs without dependency on Copr or Packit Service.
+
+## Disadvantages and caveats
+
+### SRPM builds
+
+- If we were to trigger Copr builds with SRPM created in GitHub Action we introduce blocking action of
+  acquiring the SRPM from GitHub.
+- No cache is present, which also introduces:
+  - slower build time
+  - potential time out, since Actions are time limited (iirc 24 hrs per action)
+
+### Security concerns
+
+In case of utilizing API via GitHub Actions there is a risk of modifying actions that could result
+in leaking of the secrets. GitHub mitigates this by censoring the output of GitHub Actions, but with
+our use-case it would be still possible to (indirectly) modify scripts that are run and leak the
+secrets.
```

Comment:
Secrets shouldn't be available during pull-request workflows ... and push workflows are fully under the maintainer's control.

Reply:
This is handy but will probably cause problems with the Copr builds for PRs. How we can do this?

Reply:
Ad pull request workflow, they disable secrets by default (with an override): https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#potential-impact-of-a-compromised-runner

In ideal case I would like to split the action to multiple parts and pass the secrets to the environment only for calling packit that directly queries API.

Follow-up question:
- [ ] Do we **always** create SRPM for `packit copr-build`, **even if** there is an SRPM already present?

Reply:
>     * Do we **always** create SRPM for `packit copr-build`, **even if** there is an SRPM already present?

I think so, but if we want to change this and reuse the existing srpm, this is easy to do. Maybe as another CLI argument?